xline/src/server/command.rs

```diff
@@ -216,24 +222,36 @@ where
     async fn after_sync(&self, cmd: &Command, index: usize) -> Result<SyncResponse, ExecuteError> {
         let wrapper = cmd.request();
         self.auth_storage.check_permission(wrapper).await?;
-        self.backend.insert(
-            "meta",
-            "applied_index",
+        let mut wr_ops = vec![WriteOperation::new_put(
+            META_TABLE,
+            APPLIED_INDEX_KEY,
             index.cast::<u64>().to_le_bytes(),
-            false,
-        )?;
+        )];
         match wrapper.request.backend() {
-            RequestBackend::Kv => self.kv_storage.after_sync(wrapper).await,
-            RequestBackend::Auth => self.auth_storage.after_sync(wrapper),
-            RequestBackend::Lease => self.lease_storage.after_sync(wrapper).await,
+            RequestBackend::Kv => {
+                let (res, mut ops) = self.kv_storage.after_sync(wrapper).await?;
+                wr_ops.append(&mut ops);
+                self.backend.write_batch(wr_ops, false)?;
```

Comment:
If you want to insert a key-value pair into the backend, using the `insert` method is ok because the `insert` method will do the same thing. It is implemented by `write_batch`.

Reply:
There may be more than one write operation here, we need to batch them